Implemented Admin Delete (Soft)

Snail51 opened the pull request:
Addressed #6 

MUST NOT BE ACCEPTED UNTIL #12 IS RESOLVED

Implemented admin user-deletion process. Users are soft-deleted by marking a "deleted_at" property.
Additionally, if an admin tries to create an account the uses the same email as a previously deleted user, instead of creating a new user, that user is un-deleted and updated with the creation parameters.

Snail51 committed:
add soft delete attributes

Snail51 committed:
user (soft) deletion implemented

clicking "Delete" button on view/edit user page produces admin-only users.delete view. completion of the verification "form" on this page executes users.destroy to execute the deletion

NOTE: this branch is based on (the currently not merged) `admin-create-user` branch. any changes made to that will need to be applied here

side note: i figured out how tailwind works and will retroactively apply it to said `admin-create`user` branch

Snail51 committed:
admins may ressurect previously deleted accounts

if an admin attempts to create an account that uses the email of an account that was previously soft deleted, instead of creating a new user, we un-trash that account and update it's info with the creation parameters

Snail51 committed:
move delete from to included blade doc

delete form users moved to seperate .blade.php document that is included in main `delete.blade.php` view, improving readabilty

Snail51 committed:
fix checkbox like in #12

follow proper checkbox implmentation like in https://github.com/minnesota-furs/mnfursvolunteers/pull/12#pullrequestreview-2364952142

drraccoony commented:
Please resolve merge conflicts. 💯

Snail51 committed:
Merge branch 'main' into admin-delete-soft

drraccoony commented on resources/views/users/partials/delete-user-form.blade.php:
```diff
@@ -0,0 +1,120 @@
+<section>
+    <form id="send-verification" method="post" action="{{ route('verification.send') }}">
+        @csrf
+    </form>
```
What is this doing? 🤔

Snail51 replied:
i have literally no idea, i just copied it from what was already there

Snail51 replied:
this was copied from `update-profile-information-form.blade.php` and does not appear to do anything, so they have been removed from all forms I authored. they are still present in the codebase but I'll leave removing them up to you

drraccoony commented on resources/views/users/partials/delete-user-form.blade.php:
```diff
@@ -0,0 +1,120 @@
+        @csrf
+    </form>
+
+    <form method="post" action="{{ route('users.store') }}" class="mt-6 space-y-6">
```
A post request to the `users.store` route? You sure this is right for delete user?

Snail51 replied:
this was caused by me moving the delete form into its own partial. i clearly really messed this up, as I *just didnt* change anything at all from the creation form so decided to just revert to when it was working.

drraccoony commented on resources/views/users/partials/delete-user-form.blade.php:
Why are we passing in all the edit fields again for a delete?

Snail51 replied:
because i was dumb and somehow commited a version where the delete-user-form partial was literally identical to the create-user-form partial SOMEHOW. this has been reverted to when it was working

Snail51 committed:
Revert "fix checkbox like in #12"

This reverts commit 87168102c5a3a69e007dac7c32b6c43f04e83c9e.

Snail51 committed:
Revert "move delete from to included blade doc"

This reverts commit 0b1a74b933e71751ca72db5d7981ecd401b91c71.

Snail51 committed:
Merge branch 'admin-delete-soft' of https://github.com/Snail51/mnfursvolunteers into admin-delete-soft

Snail51 committed:
fix implimentation of delete form

indentation was all over the place, this fixes it.

Snail51 committed:
reimpliment checkbox fix

matches solution of #12

Snail51 committed:
remove verification.send forms

copied from `update-profile-information-form.blade.php` i think. these do nothing in the main part of the code and are removed to improve clarity
